packages/nvidia_nat_langchain/tests/test_langchain_callback_handler.py

```
@@ -278,3 +282,76 @@ def test_extract_tools_schema_empty_and_none():
+) -> None:
+    """Push a TOOL_START event so the handler has run_id context."""
+    await handler.on_tool_start(
+        serialized={"name": tool_name},
+        input_str=input_str,
+        run_id=run_id,
+    )
+
+
+def _get_tool_end(stats: list[IntermediateStepPayload]) -> IntermediateStepPayload:
+    """Return the first TOOL_END payload from collected stats."""
+    return [s for s in stats if s.event_type == IntermediateStepType.TOOL_END][0]
```

Ruff (0.15.6)

[warning] Prefer `next(...)` over single element slice

Replace with `next(...)`

(RUF015)